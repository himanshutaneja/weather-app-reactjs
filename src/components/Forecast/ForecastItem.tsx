import { FC } from 'react';
import moment from 'moment';
import { ForecastDaily } from '../../store/types';
import { formatTemperature } from '../../helpers';
import {
  ForcastItemContainer,
  Day,
  TempMax,
  TempMin,
  WeatherIcon,
  Rain,
} from './styles';

interface ForecastItemProps {
  data: ForecastDaily;
}

const ForecastItem: FC<ForecastItemProps> = ({ data }) => {
  return (
    <ForcastItemContainer>
      <Day>{moment.unix(data.dt).format('ddd DD')}</Day>
      <TempMax>{formatTemperature(data.temp.max)}°</TempMax>
      <TempMin>{formatTemperature(data.temp.min)}°</TempMin>
      <WeatherIcon className={`wi wi-owm-${data.weather[0].id}`}></WeatherIcon>
      {data.rain && <Rain>{data.rain}%</Rain>}
    </ForcastItemContainer>
  );
};

export default ForecastItem;
