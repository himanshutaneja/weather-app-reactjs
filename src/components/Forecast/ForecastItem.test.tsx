import { shallow, ShallowWrapper } from 'enzyme';
import ForecastItem from './ForecastItem';
import { weatherForecastMock } from '../../fixtures/WeatherForecast.fixture';

describe('ForecastItem', () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    wrapper = shallow(<ForecastItem data={weatherForecastMock.daily[0]} />);
  });

  it('should render search bar correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
