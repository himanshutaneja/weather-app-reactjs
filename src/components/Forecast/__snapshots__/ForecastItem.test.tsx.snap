// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastItem should render search bar correctly 1`] = `
<styled.div>
  <styled.h3>
    Mon 19
  </styled.h3>
  <styled.div>
    15
    °
  </styled.div>
  <styled.div>
    8
    °
  </styled.div>
  <styled.i
    className="wi wi-owm-803"
  />
  <styled.div>
    11.1
    %
  </styled.div>
</styled.div>
`;
